### PR TITLE
Add certificate_data_base64 field to key_vault_certificate

### LINKED ISSUE #8072: Support for base64 output in azurerm_key_vault_certificate
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

 azurerm_key_vault_certificate outputs certificate_data as a hex string, while azuread_service_principal_certificate requires base64 format.

### New or Affected Resource(s)

* `azuread_service_principal_certificate`
* `azurerm_key_vault_certificate`

### Potential Terraform Configuration

```hcl
resource "azuread_service_principal_certificate" "example" {
  service_principal_id = azuread_service_principal.example.id
  type                 = "AsymmetricX509Cert"
  value                = azurerm_key_vault_certificate.example.certificate_data # broken - is hex string, provider expects base64
  end_date             = azurerm_key_vault_certificate.example.certificate_attribute[0].expires
}

resource "azurerm_key_vault_certificate" "example" {
  name         = "generated-cert"
```

### References

Tested with:
```
Terraform v0.12.29
+ provider.azuread v0.11.0
+ provider.azurerm v2.22.0
```

### PR DESCRIPTION
**Pull request by hazzik**
Fixes #8072

### CODE REVIEW

File: azurerm/internal/services/keyvault/key_vault_certificate_data_source.go [hunk 1]
```diff
@@ -206,6 +206,12 @@ func dataSourceKeyVaultCertificate() *schema.Resource {
 			},
 
 			"certificate_data": {
+				Type:       schema.TypeString,
+				Computed:   true,
+				Deprecated: "Deprecated in favour of `certificate_data_base64`",
```
Comment (manicminer) [inline azurerm/internal/services/keyvault/key_vault_certificate_data_source.go:211]:
```suggestion
```

File: azurerm/internal/services/keyvault/key_vault_certificate_resource.go [hunk 2]
```diff
@@ -336,6 +336,12 @@ func resourceKeyVaultCertificate() *schema.Resource {
 			},
 
 			"certificate_data": {
+				Type:       schema.TypeString,
+				Computed:   true,
+				Deprecated: "Deprecated in favour of `certificate_data_base64`",
```
Comment (manicminer) [inline azurerm/internal/services/keyvault/key_vault_certificate_resource.go:341]:
```suggestion
```

File: website/docs/d/key_vault_certificate.html.markdown [hunk 3]
```diff
@@ -56,7 +56,9 @@ The following attributes are exported:
 
 * `version` - The current version of the Key Vault Certificate.
 
-* `certificate_data` - The raw Key Vault Certificate data represented as a hexadecimal string.
+* `certificate_data` - (**Deprecated in favour of `certificate_data_base64`**) The raw Key Vault Certificate data represented as a hexadecimal string.
```
Comment (manicminer) [inline website/docs/d/key_vault_certificate.html.markdown:59]:
```suggestion
* `certificate_data` - The raw Key Vault Certificate data represented as a hexadecimal string.
```

File: website/docs/r/key_vault_certificate.html.markdown [hunk 4]
```diff
@@ -315,7 +315,8 @@ The following attributes are exported:
 * `id` - The Key Vault Certificate ID.
 * `secret_id` - The ID of the associated Key Vault Secret.
 * `version` - The current version of the Key Vault Certificate.
-* `certificate_data` - The raw Key Vault Certificate data represented as a hexadecimal string.
+* `certificate_data` - (**Deprecated in favour of `certificate_data_base64`**) The raw Key Vault Certificate data represented as a hexadecimal string.
```
Comment (manicminer) [inline website/docs/r/key_vault_certificate.html.markdown:318]:
```suggestion
* `certificate_data` - The raw Key Vault Certificate data represented as a hexadecimal string.
```

**Review by manicminer — CHANGES_REQUESTED** [review]
Hi @hazzik, thanks for this PR. There is a related PR at https://github.com/hashicorp/terraform-provider-azuread/pull/386 which also solves this, however I believe it will still be useful to export the base64 encoded certificate data here since it can be used anywhere else in Terraform configuration. Related to this, I think we should keep the existing attribute (and not deprecate it) since it's likely used by some users in contexts we're not aware of.

**Comment by hazzik** [comment]
Thanks @manicminer for the review. I'll remove depreciation.

>There is a related PR at hashicorp/terraform-provider-azuread#386 which also solves this

Unfortunately I'm using the certificate not in `azuread` provider, but in `azurerm_application_gateway` resource (so it is #8482).

**Comment by hazzik** [comment]
@manicminer changes applied.

**Review by manicminer — APPROVED** [review]
Thanks @hazzik, this LGTM

**Comment by manicminer** [comment]
Test results

Screenshot 2021-01-26 at 11 28 20

**Pull request merged**